### ISSUE TITLE
Lockfile of cron.php is waking up HDD

### ISSUE THREAD

**Issue by minils** [owncloud/core#13712]
It is not possible to spin down a HDD which is used for data storage of your ownCloud installation because of the lockfile which is written every 15 minutes to the data/ directory.
Why don't you write it somewhere else inside of the ownCloud installation?

I use a timeout of 10 minutes to spindown my HDD but it is woken up 5 minutes after it goes to sleep because of the lockfile.

(I know it is possible to change it in cron.php but maybe it should be changed or at least be configurable.)

**Comment by DeepDiver1975** [comment]
> It is not possible to spin down a HDD which is used for data storage of your ownCloud installation because of the lockfile which is written every 15 minutes to the data/ directory.
> Why don't you write it somewhere else inside of the ownCloud installation?

So - if the lock file of cron is not waking up your HDD - the jobs will do this then as they access the files within the owncloud installation (loading php files, accessing the database, accessing session data, accessing whatever cache is installed/used and finally the files living in /data) - I honestly see no point in NOT writing the lock file for cron.

@minils what would your solution to this problem look like? THX

**Comment by minils** [comment]
I thought that my configuration is somehow popular. So I thought it would be nice for others to have the option to change the location of this lock file.

I will try to explain it some more:
In my case I have the OS, the database, the webserver (including the owncloud files) installed on SD card. And the data directory is stored on SATA drive. This means that the cron job is only spinning up my drive because it is writing the lock file to the data directory which is located on the SATA drive.

If this only affects me, close the issue.

**Comment by DeepDiver1975** [comment]
@minils thanks for getting back on this issue!

This setup is indeed interesting and makes sense - furthermore I understand the annoyance of that disk spinning up for nothing.

Which database are you using and where is that data stored? THX a lot!

**Comment by minils** [comment]
I'm using mysql and the database is stored on sd card as well.

**Comment by lordaivan** [comment]
I have a similiar setup, and got the same "problem" with the HD spinning up again. Ive been thinking about that problem for a while my solution will be:
copy the data folder back onto the internal emmc card, but let the user folders (in my setup i only have a few) reside on the external hd and ln -s them back into the data folder.
I will get back to you as soon as i found time and can confirm that it works and does not have any side effects. 
Should it work, i would request putting the user folder inside an extra folder for example users, so that there is no need for manual linking the folders back into the owncloud data folder and one symbolic link is enough

**Comment by lordaivan** [comment]
I've been successful in splitting the data directory. So far the only downside was, that my owncloud client(linux) started a complete redownload of all files(might have been an error on my side, not sure) but doesn't matter. The hd spins down after 20minutes as it should, and wont spin up until i access my files.
Since i just have a couple of users and this wont change for a while im fine with creating an user then moving the dir onto the external hd and linking it back onto internal memory i'm fine with the current configuration.
But i guess that many users/admins are running their owncloud setup off an lowpower arm based device like i am, with the goal to have a low utilitys bill and low noise polution :), it would be nice to have a seperate user directory inside the owncloud data dir. Would make it a little bit more convinient.

**Comment by DeepDiver1975** [comment]
What you could do as quick workaround - specify a different path for the lock file in here:
https://github.com/owncloud/core/blob/master/cron.php#L90

**Comment by DeepDiver1975** [comment]
@minils @lordaivan please have a look at #13770 and let me know if this is a solution for your setup - THX

**Comment by minils** [comment]
It's solving the problem. But as @mmattel commented I would add some more description to the config file telling about the problem(s).

**Comment by DeepDiver1975** [comment]
we will close the issue once the pull request is merged - THX
